### ISSUE TITLE
AbortSignal is ignored when using response patching

### ISSUE THREAD

**Issue by hpohlmeyer** [mswjs/interceptors#368]
When aborting a request with an `AbortController`, fetch rejects with an `AbortError`.
This does not happen when we are using response patching at the moment.

```ts
import { FetchInterceptor } from '@mswjs/interceptors/fetch'

const interceptor = new FetchInterceptor();
interceptor.apply()

interceptor.on('request', async (request) => {
  if (request.url.endsWith("/blocked")) {
    request.respondWith(new Response("Blocked", { status: 403 }))
  }
})

// Rejects with an AbortError as expected
const controller1 = new AbortController();
fetch("/trusted", {signal: controller1.signal })
  .then(console.log, console.error);
controller1.abort();

// Fulfills with the patched response
const controller2 = new AbortController();
fetch("/blocked", { signal: controller2.signal })
  .then(console.log, console.error);
controller2.abort();
```

To keep things consistent with the fetch API, I would suggest to throw the same error when we are patching the response.

**Comment by kettanaito** [comment]
Hi, @hpohlmeyer. That's a great observation. We agree, we should not alter the original fetch in any way, which includes forwarding of whichever properties it has upon declaration (like AbortController). Thanks for opening a pull request with the fix so quickly! I will take a look when I have a minute.

**Comment by kettanaito** [comment]
## Released: v0.22.11 🎉

This has been released in v0.22.11!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.22.11)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.22.11)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
